app/repository/RestRepository.scala

```diff
@@ -1,17 +1,26 @@
 package repository
 
 import scala.concurrent.Future
-
-import repository.RestRepository.{ ErrorMessage, Row }
+import repository.RestRepository.{ ErrorMessage, Field, Row, RowKey }
 
 trait RestRepository {
-  def findRow(table: String, rowKey: String, columnFamily: String): Future[Either[ErrorMessage, Option[Row]]]
+  def findRow(table: String, rowKey: RowKey, columnFamily: String): Future[Either[ErrorMessage, Option[Row]]]
   def findRows(table: String, query: String, columnFamily: String): Future[Either[ErrorMessage, Seq[Row]]]
+
+  def update(table: String, rowKey: RowKey, beforeField: Field, afterField: Field): Future[UpdateResult]
```

Comment:
There is actually an important distinction here - but I appreciate that it may not be obvious at first glance.

The code above this in the call stack does use the language from/to as that is what the user request is about.  However, at the implementation level of the repository the "beforeField" does not actually have to be the same column as the "afterField".  Each of the arguments is typed as a Field which is a columnName / value pair.  If we stored timestamps or version number fields in the database the current implementation would allow those to be used for concurrency control (a common strategy).

i.e. the current code supports the equivalent of:
```
UPDATE table
SET afterColumn = afterValue
WHERE beforeColumn = beforeValue
```
where afterColumn != beforeColumn

Currently we do not have an explicit concurrency control column, and so we use the original value as the check value i.e beforeColumn = afterColumn.  But there did not seem to be an advantage to constraining ourselves to always work that way.

I think I originally called these checkField / targetField.  Perhaps in retrospect, they were better names?  Note that if we do want to constrain ourselves to always only using the same column for the check and the update, then the function signature should not accept two Fields, but rather a columnName and two values.